examples/cpp/xtypes/main.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file main.cpp
+ *
+ */
+
+#include <csignal>
+#include <cstdlib>
+#include <stdexcept>
+#include <thread>
+
```

Comment:
For fixing windows compilation it may just be needed `#include<functional` but to assure, if we include  `#include <fastdds/dds/domain/DomainParticipantFactory.hpp>` then windows compilation is not complaining.